### PR TITLE
make audit_log_enabled configurable

### PR DESCRIPTION
**Pull request by mcrors**
This fixes https://github.com/spiffe/helm-charts-hardened/issues/239

Added auditLogEnabled to spire-server/values.yaml.
Updated spire-server configmap.
Generated new documentation.
Tested manually against kind cluster.

### CODE REVIEW

**Review by kfox1111 — APPROVED** [review]
Looks good to me. Thanks! :)

**Comment by kfox1111** [comment]
Actually, there is one problem. The DCO needs to be fixed.

**Comment by mcrors** [comment]
Yeah, just noticed. Will fix and repush. Thanks.

File: charts/spire/charts/spire-server/README.md [hunk 1]
```diff
@@ -133,6 +133,7 @@ In order to run Tornjak with simple HTTP Connection only, make sure you don't cr
...
 | `dataStore.sql.externalSecret.key`                                                           | The key of the secret object whose value is the dataStore.sql password                                                                                                                                                                                                                                                                                               | `""`                                                                                           |
+| `auditLogEnabled`                                                                            | Set if audit log is enabled, valid values are true or false                                                                                                                                                                                                                                                                                                          | `false`                                                                                        |
```
Comment (faisal-memon) [inline charts/spire/charts/spire-server/README.md:136]:
```suggestion
| `auditLogEnabled`                                                                            | If true, enables audit logging                                                                                                                                                                                                                                                                                                          | `false`                                                                                        |
```

File: charts/spire/charts/spire-server/values.yaml [hunk 2]
```diff
@@ -173,6 +173,8 @@ dataStore:
       ## @param dataStore.sql.externalSecret.key The key of the secret object whose value is the dataStore.sql password
       key: ""
 
+## @param auditLogEnabled Set if audit log is enabled, valid values are true or false
```
Comment (faisal-memon) [inline charts/spire/charts/spire-server/values.yaml:176]:
```suggestion
## @param auditLogEnabled If true, enables audit logging
```

File: charts/spire/charts/spire-server/README.md (hunk 1 above)
Comment (faisal-memon) [reply charts/spire/charts/spire-server/README.md:136]: Disregard, forgot this is auto gen

**Pull request merged**